### ISSUE TITLE
OAI toolchain: provide options for rate limiting

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/mik#339]
The OAI-PMH harvester MIK uses, https://github.com/caseyamcl/phpoaipmh, doesn't provide any built-in rate limiting options like pausing, etc, but its README points out that https://github.com/guzzle/retry-subscriber can be used for this sort of thing. We should add rate-limiting options so brittle OAI-PMH providers don't crash as often when we point MIK at them.

**Comment by mjordan** [comment]
@bondjimbond can we close this since we have #344?

**Comment by bondjimbond** [comment]
The fix for #344 is related but different - but perhaps good enough for now? Are there other use cases for rate limiting specifically that aren't covered in #344?

**Comment by mjordan** [comment]
Fair enough - but AFAIK implementing this will be non-trivial since we'll need to dig into the OAI-PMN harvester. Let's leave it open in case someone gets the time to take a closer look.

I can't think of a more specific use case for rate limiting on the OAI harvest other than to not swamp the OAI provider, but as I understand it, the whole reason they included resumption tokens was to allow for a built-in rate limit, or at least, a limit on how many records were returned at one time. Fetching the files is outside the scope of OAI-PMH so I don't think that was a consideration when they thought about resumption tokens.
